src/hxtsdgen/Generator.hx

```diff
@@ -180,6 +191,41 @@ class Generator {
         });
     }
 
+    function generateEnumDeclaration(t:ClassType):String {
+        // TypeScript `const enum` are pure typing constructs (e.g. don't exist in JS either)
+        // so it matches Haxe abstract enum well.
+
+        // Sanitize enum name
+        var name = t.name.replace("_Impl_", "");
```

Comment:
This is fragile :) I think you need to retrieve the abstract from `KAbstractImpl` class kind and use that for the name.

Reply:
Not sure how

Reply:
Found it